src/waveform/renderers/waveformmark.cpp

```diff
@@ -57,6 +62,7 @@ void WaveformMark::setup(const QString& group, const QDomNode& node,
 // TODO(XXX): subclass and override WaveformMark::setup
 void WaveformMark::setKeyAndIndex(const ConfigKey& key, int i) {
     DEBUG_ASSERT(m_pPointCos == NULL);
-    m_pPointCos = new ControlProxy(key);
+    m_pPointCos = std::make_unique<ControlProxy>(key);
     m_text = m_text.arg(i);
+    m_iIndex = i;
```

Comment:
This function reveals that m_pPointCos and m_iIndex refer to entities outside the context of WaveformMark. WaveformMark should be split into two classes. You need a simple value object class that represents the properties of a marker by encapsulating text, color, and image. Another class is needed for associating waveform marks with actual cue points.